File: App/lib/models/login_response_dto.dart
class LoginResponseDTO {
  final int id;
  final String firstName;
  final String lastName;
  final String email;
  final String role;
  final int status;
  final String message;

  const LoginResponseDTO({
    required this.id,
    required this.firstName,
    required this.lastName,
    required this.email,
    required this.role,
    required this.status,
    required this.message,
  });

  factory LoginResponseDTO.fromJson(Map<String, dynamic> json) {
    return LoginResponseDTO(
      id: json['id'],
      firstName: json['firstName'],
      lastName: json['lastName'],
      email: json['email'],
      role: json['role'],
      status: json['status'],
      message: json['message'],
    );
  }
}

File: Api/flutter-auth-api/src/main/java/fr/esgi/flutter_auth_api/dto/response/LoginResponseDTO.java
package fr.esgi.flutter_auth_api.dto.response;

public class LoginResponseDTO {
    public Long id;
    public String firstName;
    public String lastName;
    public String email;
    public String role;
    public int status;
    public String message;

    public LoginResponseDTO(Long id, String firstName, String lastName, String email, String role, int status, String message) {
        this.id = id;
        this.firstName = firstName;
        this.lastName = lastName;
        this.email = email;
        this.role = role;
        this.status = status;
        this.message = message;
    }
}
